src/Essentials/src/Share/Share.ios.cs

```diff
@@ -36,14 +43,16 @@ Task PlatformRequestAsync(ShareTextRequest request)
 					activityController.PopoverPresentationController.SourceRect = request.PresentationSourceBounds.AsCGRect();
 			}
 
-			return vc.PresentViewControllerAsync(activityController, true);
+			await vc.PresentViewControllerAsync(activityController, true);
+			await src.Task;
 		}
 
-		Task PlatformRequestAsync(ShareFileRequest request) =>
-			PlatformRequestAsync((ShareMultipleFilesRequest)request);
+		async Task PlatformRequestAsync(ShareFileRequest request) =>
```

Comment:
This shouldn't need the async or await in it?

Reply:
Whoops totally right! Removed